swagger.yaml

```diff
@@ -5321,6 +5321,11 @@ definitions:
     properties:
       type:
         $ref: '#/definitions/platform_type'
+      is_external:
+        type: boolean
+        default: false
+        x-nullable: true
+        description: Indicates if the underlying platform type is external (read-only).
```

Comment:
@avishayt @eliorerz @gamli75

As discussed last time, I created a flag `is_external` under the `platform` key, let me know that makes sense.

Also, since it's a read-only field, I plan to ignore any potential value set by the user on cluster create or update. The of `is_external` will be always refreshed against the platform type, is that an acceptable behavior?

Reply:
Are we sure that we want this field to be read-only? 
If the user will set `oci` as platform type I thought that in this case he will must specify that this platform is external, else he might be "surprised" later when he will need to add custom manifest, also on the install-config the type will be `external` and not `oci` as expected.

Reply:
1. @eliorerz: Why make the user set the property?  What set it to "false"?  The idea behind the property was that the service is indicating to the user (and the UI, and possibly future validations) that the platform requires adding manifests.
2. @eliorerz: Is the fact that the install-config has a different platform name concerning?  In my opinion, that file is an internal implementation detail.  Is this not always the case?
3. @adriengentil: If a property is read-only, the API shouldn't allow setting it.  We should have one object for create/update and one for "GET".

Reply:
> Why make the user set the property? What set it to "false"? The idea behind the property was that the service is indicating to the user (and the UI, and possibly future validations) that the platform requires adding manifests.

Selecting OCI platform type implies the use of the external platform. Right now, I don't see a use case where it would be valid to have `platform.type = oci` and `platform.is_external = false`.

BTW, uploading manifests is not something we will enforce in the code yet, mainly because the first user will be Oracle and we want to let them experiment, also it's a bit complicated to check the content of the manifests is actually deploying a Cloud Controller Manager. We may revise this decision later.

> If a property is read-only, the API shouldn't allow setting it. We should have one object for create/update and one for "GET".

That makes a lot of sense, thank you!

Reply:
> Actually, swagger provides a [property to make a field read only](https://swagger.io/docs/specification/data-models/data-types/), that should be enough, no?

Interesting, I hadn't seen that.  We've never used it before in the project, but you're welcome to try it and see if our code generation works with it.

Reply:
> edit: Actually, swagger provides a [property to make a field read only](https://swagger.io/docs/specification/data-models/data-types/), that should be enough, no?

I'm not sure that this is available on [swagger v2](https://swagger.io/specification/v2/)

Reply:
> 1. Why make the user set the property

Mainly for 2 reasons:

- On creation the user acknowledge that this is an external platform 
- We theoretically can create vsphere (or other platforms) as external platform

However, I think that the second reason can be handled in the future if needed.

Reply:
> On creation the user acknowledge that this is an external platform
I would rather tell the user (and UI) then have the user tell us

> We theoretically can create vsphere (or other platforms) as external platform
I agree that this can be handled in the future if needed.

Reply:
> Interesting, I hadn't seen that. We've never used it before in the project, but you're welcome to try it and see if our code generation works with it.

tested it, and it works well!